### PR TITLE
Declare approve-always scopes on effect declarations (@always)

### PR DESCRIPTION
**Pull request by egonSchiele**
## What this does

Every stdlib effect now declares, on its own `effect` line, which payload fields an "approve always here" rule pins:

```
@always(name)
effect std::env { name: string }

@alwaysUnder(dir)
effect std::read { dir: string, filename: string, offset: number, limit: number }
```

`@always` pins the exact value. `@alwaysUnder` pins the value and everything under it (`{value,value/**}`). `std::policy` reads the declaration at runtime, and the agency agent's `ALWAYS_FIELDS` table is gone.

Before: asking the agent for an env var showed only approve once / approve always (every `std::env`) / reject, because `std::env` was missing from the agent's table. Now it offers `approve always here (name=BRAVE_API_KEY)`.

## How it fits together

- `lib/utils/alwaysTag.ts` reads the two tags into `ScopedField[]`. `lib/utils/tagsAbove.ts` pairs standalone tag nodes with the declaration below them without mutating, because the symbol table and typechecker run before the preprocessor attaches tags.
- The typechecker checks fields exist, tags appear once, arguments are identifiers, tagged declarations agree, and tags sit on effect declarations (`AG3019`–`AG3023`, each with an `agency explain` entry). An untagged redeclaration of a tagged effect (the guide shows users writing `effect std::read {...}`) inherits the scope and is not an error.
- Codegen emits one `__registerAlwaysScope(...)` per tagged declaration at module JS-load. `lib/runtime/alwaysScope.ts` is the process-wide registry, never checkpointed.
- Interrupts forwarded from a subprocess (`std::agency` `run` / `runFile` / `testFile`, which `std::toolbox` uses) carry their scope on the IPC message, and the parent registers it on receipt. Without this, an effect the parent never imported would silently lose the option.
- `std::policy`: `alwaysScopeFor`, `defaultScopedFields`; `buildScopedMatch`, `recordScopedRule`, and `cliPolicyHandler` take `fields` as an optional override (an entry replaces the declared scope; an empty list turns the option off). Generated rule values are escaped, so approving `ls *.md` saves a rule for that exact command. `escapeGlob` moved to `lib/runtime/policy.ts` and is shared with the built-in policies.
- Interrupts that expect a value (a question, a review) get only once answers; "always" has no meaning for them.
- `std::openUrl` now carries `host`, so "always here" covers a site rather than one URL. `std::readBinary` and `std::writeBinary` were raised but never declared; they are declared now.

## Things to look at

- `agency doc` prints the tag line above each effect declaration, so the regenerated `docs/site/stdlib/*.md` pages show the scope. The spec said `agency doc` would be unchanged; this seems better than hiding it, but it is your call.
- `make` regenerated `docs/site/diagnostics/*.md` for the five new codes.
- The agent-side change (removing `ALWAYS_FIELDS`) is covered by `lib/agents/agency-agent/tests/envPolicy.agency`, which installs `cliPolicyHandler` the way the agent now does and reads the prompt choices. I did not run the interactive agent by hand.
- `lib/utils/alwaysTag.stdlib.test.ts` holds the decision table, one row per stdlib effect; a new effect fails it until a row is added. Two rows differ from the spec: `std::createEvent` and `std::updateEvent` carry no `calendarId` in their payload, so they pin nothing.

## Tests

New Agency tests: `policy-always-registry`, `effect-always-registry` (reads the registry from a `static const`, so at module init), `always-scope-over-ipc` (the parent never declares the child's effect; fails with the IPC registration disabled, checked by hand), `policy-ask-value-interrupt`, `open-url-host`, agent `envPolicy`. Existing `policy-*` and agent policy tests pass. Unit tests cover the tag reader, the typechecker rules, the registry, `escapeGlob`, `_urlHost`, and the formatter keeping tags on effect declarations.

Spec: `packages/agency-lang/2026-08-29-always-scope-spec.md`. Plan: `packages/agency-lang/2026-08-29-always-scope-plan.md`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Cy7UTqjpTU2hoQNPmS8MCV

### CODE REVIEW

File: packages/agency-lang/docs/dev/agents/approval-policies.md [hunk 1]
```diff
@@ -21,6 +21,17 @@ In a non-interactive run (`-p`), an effect no rule decides is
 auto-rejected with an explanatory message (`stdlib/policy.agency`) — there
 is no one to ask.
 
+## What "approve always here" pins
```
Comment (egonSchiele) [inline packages/agency-lang/docs/dev/agents/approval-policies.md:24]: Read through this doc and any other docs you added to docs/dev/, as well as all the comments you have added across all files in this PR, and consider whether this is useful information for future readers to know, or whether these are just your implementation notes and thought process. If it is the latter, then they do not belong in this PR.

File: packages/agency-lang/lib/utils/tagsAbove.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if (node.type === "tag") {
+      pending = [...pending, node];
+      continue;
+    }
+    out.push({ node, tags: pending });
+    pending = [];
+    const body = (node as { body?: unknown }).body;
+    if (Array.isArray(body)) {
+      out.push(...tagsAbove(body as AgencyNode[]));
+    }
```
[the remark this answers is not in the document]
Comment (egonSchiele) [reply packages/agency-lang/lib/utils/tagsAbove.ts]: Fixed in 3083d39c6: tagsAbove now recurses through bodySlots(node), and a new typechecker test declares a tagged effect inside an if branch and expects the unknown-field diagnostic.

File: packages/agency-lang/stdlib/policy.agency [hunk 3]
```diff
@@ -246,26 +250,51 @@ export def validatePolicy(policy: Record<string, any>): Result<void> {
...
+  """Every declared always-scope, keyed by effect. What `cliPolicyHandler` uses when its caller passes no `fields`."""
+  return _allAlwaysScopes()
+}
+
+// The one resolution rule: an explicit entry wins (an empty list
+// suppresses the option); otherwise the effect's declared scope.
+def scopedFieldsFor(effect: string, fields: ScopedRuleFields): ScopedField[] {
+  if (fields[effect] != undefined) {
+    return fields[effect]
+  }
```
[the remark this answers is not in the document]
Comment (egonSchiele) [reply packages/agency-lang/stdlib/policy.agency]: Fixed in 3083d39c6: scopedFieldsFor checks keys(fields).includes(effect), so a plain {} override never resolves a prototype member.

File: packages/agency-lang/lib/runtime/ipc.ts [hunk 4]
```diff
@@ -878,6 +878,9 @@ function attachStdoutForwarder(
 
 async function handleInterruptMessage(s: RunSession, msg: any): Promise<void> {
   const { effect, message, data, origin, expectsValue } = msg.interrupt;
+  // The child carries the scope its effect declaration gave it; this process
+  // may never have imported that module. An empty scope registers nothing.
+  __registerAlwaysScope(effect, msg.interrupt.alwaysScope ?? []);
   try {
```
[the remark this answers is not in the document]
Comment (egonSchiele) [reply packages/agency-lang/lib/runtime/ipc.ts]: Fixed in 3083d39c6, and made stricter than moving it inside the try. The parent now calls adoptAlwaysScope, which never throws: it registers a child scope only for an effect this process does not already know, and ignores a malformed value. A subprocess is untrusted input, so it can fill a gap but never replace the scope from this process own declarations. The call also sits inside the try. On top of that, buildScopedMatch returns {} when any scoped field is missing from the interrupt data, the prompt offers approve-always-here only for a non-empty match, and recordScopedRule writes nothing for an empty match, because an empty match is a catch-all approve in matchesRule. Tests cover each of those.

File: packages/agency-lang/lib/stdlib/system.ts [hunk 5]
```diff
@@ -171,3 +171,10 @@ export async function _screenshot(
 export function _setTitle(title: string): void {
   process.title = title;
 }
+
+/** The hostname of a URL, or "" when the string is not a URL. Used as the
+ *  policy scope for std::openUrl. `URL.canParse` avoids a try/catch that
+ *  would swallow the error. */
+export function _urlHost(url: string): string {
+  return URL.canParse(url) ? new URL(url).hostname : "";
```
[the remark this answers is not in the document]
Comment (egonSchiele) [reply packages/agency-lang/lib/stdlib/system.ts]: Fixed in 3083d39c6: _urlHost returns the whole string when the URL has no hostname (mailto:, file:) or does not parse, so each such value pins only itself. Test updated.

File: packages/agency-lang/docs/dev/agents/approval-policies.md (hunk 1 above)
Comment (egonSchiele) [reply packages/agency-lang/docs/dev/agents/approval-policies.md:24]: Trimmed in 3083d39c6. Removed from effect-always-tag.md: the paragraph on the old ALWAYS_FIELDS table, the untested-resume note, and the coverage-test detail about duplicate declarations. Removed the test-only remark from the IpcInterruptMessage comment. What is left in the dev note is the pipeline, the per-process registry and the IPC hand-off, the empty-match guard, and the untagged-redeclaration rule; the approval-policies section is one paragraph saying where the pinned fields come from and what the fields override does.
